martin/src/cog/errors.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::PathBuf;
+
+use png::EncodingError;
+use tiff::TiffError;
+
+#[derive(thiserror::Error, Debug)]
+pub enum CogError {
+    #[error("Couldn't decoded {1} as tiff file: {0}")]
+    InvalidTifFile(TiffError, PathBuf),
+
+    #[error("Requested zoom level:{0} from file {1} is  out of range, the zoom level is from {2} to {3}")]
+    ZoomOutOfRange(u8, PathBuf, u8, u8),
+
+    #[error("Couldn't find any image(the tiff tag newsubfile is not mask) in the tiff file: {0}")]
+    NoImagesFound(PathBuf),
+
+    #[error("Couldn't seek to ifd number {1} (0 based indexing) in tiff file {2} : {0}")]
+    IfdSeekFailed(TiffError, usize, PathBuf),
+
+    #[error("Too many images in the tiff file: {0}")]
+    TooManyImages(PathBuf),
+
+    #[error("Couldn't find tags {1:?} at ifd {2} of tiff file {3} : {0}")]
+    TagsNotFound(TiffError, Vec<u16>, usize, PathBuf),
+
+    #[error("Planar configuration not equals to 1 is not supported, the tiff file is {2}")]
+    PlanaConfigurationNotSupported(PathBuf, usize, u16),
```

Comment:
typo, please correct the usage as well ^^

```suggestion
    PlanarConfigurationNotSupported(PathBuf, usize, u16),
```